src/data-table/index.js

```diff
@@ -105,25 +103,30 @@ class DataTable extends Component {
   }
 
   columns = () => {
-    const { children, columns, data, zeroConfig } = this.props
+    const { children, columns, data} = this.props
+    const emptyData = []
+    if (!children && !columns) {
+      const columns = []
+      if (data.length > 0) {
+        Object.keys(data[0]).forEach(key => (key !== '_id' && columns.push({
+          ...columnDefaultProps,
+          name: key,
+          dataKey: key,
+          pickable: true,
+          searchable: true,
+          sortType: Number.isInteger(data[0][key]) ? 'basic' : 'string'})))
+        return columns
+      }
+      else {
```

Comment:
With early return, you don't need explicit `else`.

It's also a good practice to apply early return for the "minor" case (least logic):

```js
if (!data.length) {
  return emptyData
}
// ... otherwise logic
```

This way, the "major" case (most logic) gets less indentation, making the code a bit easier on readability.